### PR TITLE
[DOCS] Clarify Recovery Settings for Shard Relocation

### LINKED ISSUE #14539: Clarify recovery vs. relocation in documentation
Somewhere in our documentation, it will be nice to clarify that relocation is a form/type of recovery, so that the [recovery settings](https://www.elastic.co/guide/en/elasticsearch/reference/master/recovery.html#recovery) actually apply to both shard recoveries and relocations (even though the parameter name of these settings refers to "recovery" only).

### PR DESCRIPTION
**Pull request by jrodewig**
- Adds a sentence clarifying that recovery settings also apply to shard relocation
- Rewrites introduction to highlight when peer recovery occurs
- Rewrites and reformats setting descriptions for scannability and clarity

Resolves #14539 🤞

### CODE REVIEW

**Comment by jrodewig** [comment]
please retest this

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
Hi @ppf2

Let me know if this addresses your concerns from #14539.

I also included some other rewrites for clarity, but I can break those out into another PR if wanted. Thanks!

File: docs/reference/modules/indices/recovery.asciidoc [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
+existing shard copy.
 
-The following _expert_ setting can be set to manage the resources consumed by
-peer recoveries:
+Peer recovery automatically occurs when {es}:
+
+* Recreates a shard lost during node failure
+* Relocates a shard to another node due to a cluster rebalance or changes to the
+<<modules-cluster, shard allocation settings>>
+* Restores a <<modules-snapshots, snapshot>>
```
Comment (ywelsch) [inline docs/reference/modules/indices/recovery.asciidoc]: recovery from snapshot is different to peer recovery and has different settings

File: docs/reference/modules/indices/recovery.asciidoc [hunk 2]
```diff
@@ -1,33 +1,39 @@
...
+
+[float]
+==== Peer recovery settings
+You can use the following _expert_ settings to manage resources for peer
```
Comment (ywelsch) [inline docs/reference/modules/indices/recovery.asciidoc]: The first setting (`indices.recovery.max_bytes_per_sec`) is not an expert-level setting and is quite often adjusted. Let's separate it from the second one which should be attributed as expert-level and will very rarely need adjustment.

**Review by ywelsch — CHANGES_REQUESTED** [review]
While we're already touching these docs here, maybe cross-link to `[[cat-recovery]]` which provides progress information about recoveries.

**Comment by jrodewig** [comment]
Thanks for the review @ywelsch! I addressed your feedback in commit d09877e.

Let me know if you have any other concerns.

**Comment by jrodewig** [comment]
retest this please

**Comment by jrodewig** [comment]
@ywelsch Let me know if these changes look okay to you.

Happy to close or rewrite if needed.

**Review by ywelsch — APPROVED** [review]
Great, thank you! Sorry for the late reply here, I was on vacation.

**Comment by jrodewig** [comment]
No worries! Thank you!

**Pull request merged**